From 251a495d2a72de7e892ade7f64980f51f2fcc0dd Mon Sep 17 00:00:00 2001
From: elif dede <edede@twitter.com>
Date: Mon, 17 Nov 2014 16:53:08 -0800
Subject: [PATCH] PARQUET-135: Input location is not getting set for the
 getStatistics in ParquetLoader when using two different loaders within a Pig
 script.

Author: elif dede <edede@twitter.com>

Closes #86 from elifdd/parquetLoader_error_PARQUET-135 and squashes the following commits:

b0150ee [elif dede] fixed white space
bdb381a [elif dede] PARQUET-135: Call setInput from getStatistics in ParquetLoader to fix ReduceEstimator errors in pig jobs
---
 .../format/converter/TestParquetMetadataConverter.java       | 2 +-
 parquet-pig/src/main/java/parquet/pig/ParquetLoader.java     | 5 +++--
 2 files changed, 4 insertions(+), 3 deletions(-)

diff --git a/parquet-hadoop/src/test/java/parquet/format/converter/TestParquetMetadataConverter.java b/parquet-hadoop/src/test/java/parquet/format/converter/TestParquetMetadataConverter.java
index 0f7dccf675..367f5d33cb 100644
--- a/parquet-hadoop/src/test/java/parquet/format/converter/TestParquetMetadataConverter.java
+++ b/parquet-hadoop/src/test/java/parquet/format/converter/TestParquetMetadataConverter.java
@@ -244,7 +244,7 @@ public void randomTestFilterMetaData() {
       try {
         verifyAllFilters(metadata(rgs), splitSize);
       } catch (AssertionError e) {
-        throw new AssertionError("fail verifyAllFilters(metadata(" + Arrays.toString(rgs) + "), " + splitSize + ")", e);
+	  throw new AssertionError("fail verifyAllFilters(metadata(" + Arrays.toString(rgs) + "), " + splitSize + ")", e);
       }
     }
   }
diff --git a/parquet-pig/src/main/java/parquet/pig/ParquetLoader.java b/parquet-pig/src/main/java/parquet/pig/ParquetLoader.java
index 2e96ecf5d1..7bf70e9464 100644
--- a/parquet-pig/src/main/java/parquet/pig/ParquetLoader.java
+++ b/parquet-pig/src/main/java/parquet/pig/ParquetLoader.java
@@ -287,8 +287,9 @@ private boolean isElephantBirdCompatible(Job job) {
   public ResourceStatistics getStatistics(String location, Job job)
       throws IOException {
     if (DEBUG) LOG.debug("LoadMetadata.getStatistics(" + location + ", " + job + ")");
-    // We do not need to call setInput
-    // as setLocation is guaranteed to be called before this
+    /* We need to call setInput since setLocation is not
+       guaranteed to be called before this */
+    setInput(location, job);
     long length = 0;
     try {
       for (InputSplit split : getParquetInputFormat().getSplits(job)) {
